### ISSUE TITLE
Kernel constructor is incompatible with Xeus 2

### ISSUE THREAD

**Issue by egnha** [jupyter-xeus/xeus-calc#31]
**Issue**: The current main.cpp is incompatible with the new kernel constructor in Xeus 2. (I presume xeus-calc ought to be current with xeus.)

**Fix**: Apply the changes in the Xeus 2 example [main.cpp](https://github.com/jupyter-xeus/xeus/blob/master/example/src/main.cpp), e.g., 2937cc6855620f9b9f217b6c143363d0487cc7c7.

**Sample build error**:
```
$ make
[ 25%] Building CXX object CMakeFiles/xeus-calc.dir/src/xeus_calc_interpreter.cpp.o
[ 50%] Linking CXX shared library libxeus-calc.dylib
[ 50%] Built target xeus-calc
[ 75%] Building CXX object CMakeFiles/xcalc.dir/src/main.cpp.o
.../xeus-calc/src/main.cpp:28:19: error: no matching constructor for initialization of 'xeus::xkernel'
    xeus::xkernel kernel(config, xeus::get_user_name(), std::move(interpreter));
                  ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.../include/xeus/xkernel.hpp:31:20: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
    class XEUS_API xkernel
                   ^
.../include/xeus/xkernel.hpp:62:9: note: candidate constructor not viable: requires at least 4 arguments, but 3 were provided
        xkernel(const std::string& user_name,
        ^
.../include/xeus/xkernel.hpp:51:9: note: candidate constructor not viable: requires at least 5 arguments, but 3 were provided
        xkernel(const xconfiguration& config,
        ^
1 error generated.
make[2]: *** [CMakeFiles/xcalc.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/xcalc.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by egnha** [comment]
Additionally, the xeus version in various config files (e.g., for conda and CI) would require updating.

**Comment by JohanMabille** [comment]
This has been fixed in #34
